src/components/TagDropdown.tsx

```diff
@@ -0,0 +1,156 @@
+
+interface TagDropdownProps {
+    tags: string[];
+}
+
+export default function TagDropdown(props: TagDropdownProps) {
+    const [display, setDisplay] = useState(false);
+    const [selectedTags, setSelectedTags] = useState<string[]>([]);
+    const [searchInput, setSearchInput] = useState("");
+    const [filteredTags, setFilteredTags] = useState(props.tags);
+
+    const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const input_text: string = event.target.value;
+        event.preventDefault();
+        setSearchInput(input_text);
+        if (input_text.length > 0) {
+            setFilteredTags(
+                props.tags.filter((tag) => {
+                    return tag.match(input_text);
+                })
+            );
+        } else {
+            setFilteredTags(props.tags);
+        }
+    };
+
+    // Handles actions when 'Enter' key is pressed
+    const handleEnter = (key_event: React.KeyboardEvent) => {
+        if (key_event.key == "Enter") {
+            addTag(searchInput);
+        }
+    }
+
+    const handleAdd = () => {
+        addTag(searchInput);
+    }
+
+    // Controls display of dropdown
+    const handleClick = () => {
+        if (!display) {
+            setDisplay(true);
+        }
+    };
+
+    // Controls tag selection
+    function handleSelectedTags(tag: string) {
+        // if tag is already selected, deselect it and remove from selectedTags
+        if (selectedTags.includes(tag)) {
+            const toRemove = selectedTags.indexOf(tag);
+            setSelectedTags(() => [
+                ...selectedTags.slice(0, toRemove),
+                ...selectedTags.slice(toRemove + 1),
+            ]);
+        } else {
+            setSelectedTags([...selectedTags, tag]);
+        }
+    }
+
+    function deselectTag(tag: string) {
+        const toRemove = selectedTags.indexOf(tag);
+        setSelectedTags(() => [
+            ...selectedTags.slice(0, toRemove),
+            ...selectedTags.slice(toRemove + 1),
+        ]);
+    }
+
+    function addTag(tag: string) {
+        // If tag is in props.tags, add tag to selectedTags array
+        // Else (new tag), add tag to props.tags (large tag array), probably requires request to db, then add tag to selectedTags
+        // ?? QUESTION: do we actually want a New tag to be added every time someone hits enter
+        if (!props.tags.includes(tag)) {
+            props.tags.push(tag);
+            // console.log("pushed tag");
+            // console.log(props.tags);
+        } 
+        setSelectedTags([...selectedTags, tag]);
+        // clear search bar
+        setSearchInput("");
+    }
+
+    return (
+        <div className="flex flex-col p-4 gap-[10px]">
+            <div className="border-2 relative">
+                {searchInput != "" && (
+                    <button 
+                        onClick={handleAdd}
+                        className="absolute right-0 py-3 px-4"
```

Comment:
The button gets added to the DOM correctly, but the font color is white so we can't see it! If you could adjust the styling so that either the background color or the font color is different that would be great. My suggestion for now would be to just add `text-blue-600` until we have the design language for the entire site ready to go.